apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/TransactionPool.cs

```diff
@@ -161,43 +150,50 @@ public static Task CommitAsync(this Transaction transaction, Session session, IE
         /// <param name="transaction"></param>
         /// <param name="session"></param>
         /// <returns></returns>
-        public static Task RollbackAsync(this Transaction transaction, Session session)
+        public static Task<bool> RollbackAsync(this Transaction transaction, Session session)
         {
             return RunFinalMethodAsync(transaction, session,
-                info => info.SpannerClient.RollbackAsync(session.GetSessionName(), info.ActiveTransaction.GetTransactionId())
-                    .WithSessionChecking(() => session));
+                async info =>
+                {
+                    await info.SpannerClient
+                        .RollbackAsync(session.GetSessionName(),
+                            info.ActiveTransaction.GetTransactionId())
+                        .WithSessionChecking(() => session).ConfigureAwait(false);
+                    return true;
+                });
         }
 
-        private static async Task RunFinalMethodAsync(Transaction transaction, Session session, Func<SessionInfo, Task> commitOrRollbackAction)
+        private static async Task<T> RunFinalMethodAsync<T>(Transaction transaction, Session session, Func<SessionInfo, Task<T>> commitOrRollbackAction)
         {
             SessionInfo info;
             if (s_sessionInfoTable.TryGetValue(session, out info))
             {
                 //we should have already waited, but extra checking cannot hurt.
                 await info.WaitForPreWarmAsync().ConfigureAwait(false);
-                if (!info.HasActiveTransaction
-                    || !Equals(info.ActiveTransaction.Id, transaction.Id))
+                if (!info.HasActiveTransaction)
+                {
+                    throw new InvalidOperationException("The transaction being committed was not found to have an entry.");
+                }
+                if (!Equals(info.ActiveTransaction.Id, transaction.Id))
```

Comment:
I'd personally use `if (info.ActiveTransaction.Id != transaction.Id)` here (using the overloaded operator). Subjective though. (Apologies if this is the place I left a comment before and I just don't remember the response...)